pkg/plugins/document/boltdb/boltdb.go

```diff
@@ -359,6 +353,61 @@ func (s *BoltDocService) Query(collection *document.Collection, expressions []do
 	}, nil
 }
 
+func (s *BoltDocService) Query(collection *document.Collection, expressions []document.QueryExpression, limit int, pagingToken map[string]string) (*document.QueryResult, error) {
+	newErr := errors.ErrorsWithScope(
+		"BoltDocService.Query",
+		map[string]interface{}{
+			"collection": collection,
+		},
+	)
+
+	return s.query(collection, expressions, limit, pagingToken, newErr)
+}
+
+func (s *BoltDocService) QueryStream(collection *document.Collection, expressions []document.QueryExpression, limit int) document.DocumentIterator {
+	newErr := errors.ErrorsWithScope(
+		"BoltDocService.QueryStream",
+		map[string]interface{}{
+			"collection": collection,
+		},
+	)
+
+	var tmpLimit = limit
+	res, fetchErr := s.query(collection, expressions, limit, nil, newErr)
+
+	// Initial fetch
+	var documents []document.Document = res.Documents
+	var pagingToken = res.PagingToken
```

Comment:
If a fetch error occurs here, what would `res.Documents` return? Could it result in a nil pointer dereference and panic?

Reply:
Have put in an error only iterator in the initial state fetch case, to avoid this.